src/odemis/driver/lakeshore.py

```diff
@@ -108,11 +103,74 @@ def __init__(self, name, role, port, sensor_input='b', output_channel=2, **kwarg
         except LakeshoreError as ex:
             logging.warning("Discarding initial error status: %s", ex)
 
-        # Vigilant attributes of the controller.
-        self.temperature = FloatVA(unit=u"°C", value=self.GetSensorTemperature(), readonly=True)
-        self.targetTemperature = FloatContinuous(value=self.GetSetpoint(), unit=u"°C", range=[-273, 50], setter=self._set_targetTemperature)
-        self.heating = IntEnumerated(value=self.GetHeaterRange(), choices={0: "Off", 1: "Low", 2: "Medium", 3: "High"},
-                                     setter=self._set_heating)
+        # set supported sensor input and output channel values
+        if md == "MODEL335":
+            supported_sensor_inputs = ('A', 'B')
+            supported_output_channels = (1, 2)
+        elif md == "MODEL350":
+            supported_sensor_inputs = ('A', 'B', 'C', 'D')
+            supported_output_channels = (1, 2, 3, 4)
+        else:
+            raise IOError("Incompatible device model detected.")
+
+        # Check if there's only one sensor input and the input is defined as string
+        if isinstance(sensor_input, str):
+            sensor_input = {"": sensor_input}
+
+        # Check if there's one or multiple sensor input(s) and defined as dictionary
+        elif not isinstance(sensor_input, dict):
+            raise ValueError("Invalid sensor input type. Sensor input must be either str or dict")
+
+        self._sensor_inputs = {k: v.upper() for k, v in sensor_input.items()}
+        for va_name, sensor_name in self._sensor_inputs.items():
+            if sensor_name not in supported_sensor_inputs:
+                raise ValueError(f"Sensor input must be one of the following: {supported_sensor_inputs}")
+
+            # Vigilant attributes of the temperature.
+            va = FloatVA(unit=u"°C", value=self.GetSensorTemperature(sensor_name), readonly=True)
+            if va_name == "":
+                full_va_name = "temperature"
+            else:
+                full_va_name = f"{va_name}Temperature"
+            setattr(self, full_va_name, va)
+
+        # Check if there's only one int type output channel
+        if isinstance(output_channel, int):
+            output_channel = {"": output_channel}
+        elif not isinstance(output_channel, dict):
+            raise ValueError("Invalid output channel type. Output channel must be either str or dict")
+
+        self._setters = []
```

Comment:
Add a comment to explain why it's there. Something like "Keep reference to the partial functions, to avoid them being garbage collected".